alerting/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/resthandler/RestIndexMonitorAction.kt

```diff
@@ -180,6 +184,36 @@ class RestIndexMonitorAction(
             }
         }
 
+        /**
+         * This function checks whether the [Monitor] has an [HttpInput] with localhost. If so, make sure the port is same as specified in settings.
+         */
+        private fun validateLocalPort(monitor: Monitor, settingsPort: Int) {
+            if (monitor.inputs.isNotEmpty() && monitor.inputs.single() is HttpInput) {
+                val httpInput = monitor.inputs.single() as HttpInput
+                // Build url field by field if not provided as whole.
+                val constructedUrl = if (Strings.isEmpty(httpInput.url)) {
```

Comment:
there is similar implementation in `HttpExtensions.kt.HttpInput.toGetRequest()` that uses different `isEmpty`. One uses `org.elasticsearch.common.Strings.isEmpty` and other uses `kotlin.String.isEmpty`. Lets be consistent on what use and even better we can extract the logic to `constructedUrl` and share function in both places.

Reply:
Created a shared function `HttpInput.toConstructedUrl()`in `HttpExtensions.kt`